### ISSUE TITLE
Feature request: forkMode always

### ISSUE THREAD

**Issue by pan3793** [scalatest/scalatest-maven-plugin#99]
it's useful for integration tests that contain lots of singletons, because it's hard to pick up all singletons and reset them if we want to change the configuration(which will affect those singleton objects) in another test.

**Comment by cheeseng** [comment]
@pan3793 Makes sense, imho we may want to support this in org.scalatest.tools.Runner, e.g. to have a server running in the main process which forks child process that use socket reporter to send back the result back to the reporters on the main process.  We have similar thing in the sbt's Framework class that's called by sbt, we can probably do similar thing in ScalaTest own Runner.

@bvenners What do think about the idea?

**Comment by pan3793** [comment]
Hi @cheeseng, is there a plan and ETA to support it? currently, we must fall back to JUnit since the maven surefire plugin support it.

**Comment by cheeseng** [comment]
@cheeseng Hmm, unfortunately I don't have a ETA for you, but I do want to get started on it, though I also struggle on what's the best next step for it.

**Comment by pan3793** [comment]
Anyway, thanks for your quick response and maintaining such a great tool.
